### PR TITLE
Support multiple gets within single Profiler run

### PR DESCRIPTION
**Pull request by mrocklin**
Sometimes one computation spans multiple calls to `get`.  This resets the profiler at the context manager's start/exit rather than get's start/finish.

### CODE REVIEW

**Comment by shoyer** [comment]
:+1: this is pretty common with xray

**Comment by mrocklin** [comment]
@jcrist any feedback on this?

File: dask/diagnostics/profile.py [hunk 1]
```diff
@@ -57,13 +61,17 @@ def _posttask(self, key, value, dsk, state, id):
...
+        self._results.clear()
+
     def results(self):
         """Returns a list containing namedtuples of:
 
         TaskData(key, task, start_time, end_time, worker_id)"""
 
-        results = dict((k, v) for k, v in self._results.items() if len(v) == 5)
-        return list(starmap(TaskData, results.values()))
+        return
```
Comment (jcrist) [inline dask/diagnostics/profile.py]: This is overwritten by an attribute. Should probably be a property somehow.

File: dask/diagnostics/profile.py [hunk 2]
```diff
@@ -43,10 +43,14 @@ class Profiler(Callback):
     """
     def __init__(self):
         self._results = {}
+        self.results = []
         self._dsk = {}
 
-    def _start(self, dsk):
+    def __enter__(self):
         self.clear()
+        return super(Profiler, self).__enter__()
```
Comment (jcrist) [inline dask/diagnostics/profile.py]: This now depends on using the contextmanager syntax, or calling `clear` explicitly after calculations if registered. Probably the only way to accomplish this, but it should be noted.

Comment (mrocklin) [inline dask/diagnostics/profile.py]: Added in the docstring

File: dask/diagnostics/profile.py (hunk 1 above)
Comment (mrocklin) [inline dask/diagnostics/profile.py]: Good catch.  Removed.  Just using the attribute directly.

**Comment by jcrist** [comment]
Looks good to me.

**Pull request merged**